### DISCUSSION TITLE
Support for Apicula compatible boards?

### DISCUSSION THREAD

**Discussion by Harvie** [FPGAwars/icestudio#814] [New Boards Requests]
Hello,
can you please add boards supported by apicula project?
https://github.com/YosysHQ/apicula

Namely:
    Trenz TEC0117: GW1NR-UV9QN881C6/I5
    Sipeed Tang Nano: GW1N-LV1QN48C6/I5
    Sipeed Tang Nano 1K: GW1NZ-LV1QN48C6/I5
    Sipeed Tang Nano 4K: GW1NSR-LV4CQN48PC7/I6
    Sipeed Tang Nano 9K: GW1NR-LV9QN88PC6/I5 
    Seeed RUNBER: GW1N-UV4LQ144C6/I5
    @Disasm honeycomb: GW1NS-UX2CQN48C5/I4

These are some of the cheapest FPGA boards available on the market right now. So it would really make sense to support them for education purposes... But even the specs are still very reasonable.

**Comment by cavearr** [comment]
Hi @Harvie ! i don't have any of this boards , i'll buy it one of them and prepare the integration along the next weeks.

I'm maintaining this thread open to inform you about the progress.

**Comment by cavearr** [comment]
Hi @Harvie ! boards are comming, asap i'll been working on it

**Comment by Harvie** [comment]
Thank you! Which one did you ordered? So far i have `Sipeed Tang Nano 9K`, so i can help you with testing...

**Comment by k0055t** [comment]
How is the development?
Are there any problems with implementing support? I also have Sipeed Tang Nano 9K, so can help with testing. I would like to program the board with opensource toolchain. Thanks for your work

**Comment by cavearr** [comment]
Hi team! i'm waiting for my board, i bought two models:

- Tang Nano  9K 
- Sipeed Lichee Tang Nano GW1N-1

In theory , boards should be arrive one week ago, but for the moment the estimated date to arrive is for August 22.

I'm waiting for it, as soon as i have it, i'm telling you, all your help , tests ,feedback and ideas in the integration process should be very useful.

Thanks a lot !

**Comment by Harvie** [comment]
Can't wait to try this :innocent:

**Comment by cavearr** [comment]
Hi @Harvie! I am very close to the apicula support! Thank you!, in this thread I will notify the initial versions, your comments and tests will be very important.

Thank you and keep this thread to talk about it.

**Comment by kami4ka** [comment]
Any news?
The boards get more popular.

**Comment by Harvie** [comment]
Fixing this would be great idea for xmas present :-D

**Comment by cavearr** [comment]
Hi tomas! With hight probability in christmas this integration will be finished 😉

**Comment by gbr1** [comment]
Hi! Is there any update? Thank you in advance

**Comment by cavearr** [comment]
I'm very near to publish the integration. Probably at the end of the month at the latest.

there's little left!

**Comment by cavearr** [comment]
Hi @Greeniac1 ! this will be very soon ready to test.  Sorry for the delay but i'm rewriting some core parts needed to this and take me more time that expected. Along  August should be ready. I'll maintain you informe here.

Thanks!

**Comment by Harvie** [comment]
@Greeniac1 haha did you came here after Great Scott featured Tang Nano 9K yesterday? :-D
https://youtu.be/BcUzuwP-WPI?t=597

**Comment by TheZ3ro** [comment]
@cavearr is there a way we can help with the development?

**Comment by cavearr** [comment]
Hi @TheZ3ro ! sorry for my long delay with this task but i'm working in a big core changes and i need to move the priority of some tasks.  Apicula support is very near to be working, i hope give you good news in the next weeks.

**Comment by karaketir16** [comment]
Are there any updates? It looks like a great tool, but I only have Tang Nano.

**Comment by cavearr** [comment]
Hi! Yes se finish the apicula integration few days algo and in a very few days we will release apicula support for icestudio.

**Reply by zamonary1** [reply]
Hello. It's been quite some time. Any updates on apicula support?

**Comment by cavearr** [comment]
Hello! I am reorganizing the issues, trying to give more visibility and be able to better organize and prioritize the work and my roadmap (I will publish it very soon), I am converting some issues that are **new feature requests** as **ideas** on Github **discussions**.

Thanks for understanding.

**Comment by zamonary1** [comment]
Hello, i am gonna get my 20k soon, has support for tang nano been added already?

**Comment by cavearr** [comment]
Hi, i'm working in a huge new version of icestudio for end of September (or start of October) in this version Apicula  should be supported. Sorry for the delay but i'm refactoring, and doing a lot of new things very important for the future of Icestudio.

**Comment by ellisgl** [comment]
@cavearr  Just wondering how it's going and if everything is OK.

**Comment by cavearr** [comment]
Hi @ellisgl Thanks for asking. I'm working really hard on a new version of IceStudio that's going to be a huge leap forward. A few months ago, I realized we had a lot of technological debt, and the way we were evolving IceStudio wasn't going to lead anywhere. We'd just stay stuck in the current rut, making little progress and constantly complicating everything.

I'm in the final stages of development. I wanted to have it ready in October, but the first beta will be released between the end of this month and December. I'll keep you updated. Besides support for new sound cards like Sipeed, there will be many changes, and I think you'll like them.

I'm staying "in the shadows" until I have a working version to avoid excessive pressure and premature feedback that would slow progress even further. You'll hear from me soon :)

Thanks again for your feedback and interest.

**Comment by ellisgl** [comment]
@cavearr  Just checking in on you. Besides having the Tang Nano and Mega boards, I just got in my IcePi Zero today. =)

**Reply by Harvie** [reply]
<img width="168" height="300" alt="image" src="https://github.com/user-attachments/assets/98b1d2ec-d12e-4233-9402-e3c38def808a" />

**Reply by ellisgl** [reply]
I'm really wondering though...

**Reply by ellisgl** [reply]
Well it looks like the nightly has it. Unfortunately I can't get it run anymore.. I'm sure it's PEBKAC.

Yup, it was a PEBCAK issue. I updated the node libraries on accident..

**Comment by cavearr** [comment]
Hi everyone! I’m currently rolling out a lot of improvements—including Gowin support. There are just a few final details to publish, and everything will be accompanied by video tutorials. Expect fresh news in just a few days; I hope you really enjoy the new phase this version is launching.

I’ll keep you posted right here over the coming days!

**Reply by Harvie** [reply]
<img width="640" height="640" alt="image" src="https://github.com/user-attachments/assets/8a077b94-0849-41c6-a366-1c02f2d9b5c5" />
